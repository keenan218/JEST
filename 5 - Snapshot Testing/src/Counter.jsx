import React, {useState} from 'react';
export default function Counter() {
    const [counter, setCounter] = useState(0);

    return (
        <div>
            <p>Counter: {counter}</p>
            <button className="increment" onClick={() => setCounter(counter => counter+1)}> Increment Count </button> 
            <button className="decrement" onClick={() => setCounter(counter => counter-1)}> Decrement Count </button>
        </div>
    )
}

