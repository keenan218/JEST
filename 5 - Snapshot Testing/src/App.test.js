import React from 'react';
import Counter from './Counter';
import renderer from 'react-test-renderer';

test("It matches the snapshot", () => {
  const tree = renderer.create(<Counter/>).toJSON();
  expect(tree).toMatchSnapshot();
})
