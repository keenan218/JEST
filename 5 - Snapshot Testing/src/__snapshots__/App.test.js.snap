// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It matches the snapshot 1`] = `
<div>
  <p>
    Counter: 
    0
  </p>
  <button
    className="increment"
    onClick={[Function]}
  >
     Increment Count 
  </button>
  <button
    className="decrement"
    onClick={[Function]}
  >
     Decrement Count 
  </button>
</div>
`;
